var/spack/repos/builtin/packages/root/package.py

```diff
@@ -31,71 +43,431 @@ class Root(CMakePackage):
+    variant('vc', default=False,
+        description='Enable Vc for adding new types for SIMD programming')
+    variant('vdt', default=True,
+        description='Enable set of fast and vectorisable math functions')
+    variant('x', default=True,
+        description='Enable set of graphical options')
+    # variant('xinetd', default=False,  - not supported by spack
+    #    description='Enable a daemon process manager')
+    variant('xml', default=True,
+        description='Enable XML parser interface')
+    variant('xrootd', default=False,
+        description='Build xrootd file server and its client')
+
+    # ################# Variants dependencies ##########################
 
+    # Davix variant also requires openssl support
+    depends_on('openssl', when='+davix')
+
+    # If not x variant, then asimage,opengl,qt4 and tiff are not needed either
+    # or: if ~x then ~asimage~opengl~qt4~tiff
+
+    # ###################### Compiler variants ########################
+
+    variant('cxxstd',
+            default='11',
+            values=('11', '14', '17'),
+            multi=False,
+            description='Use the specified C++ standard when building.')
+
+    # ###################### Dependencies ##############################
+
+    # minimum cmake version required
     depends_on('cmake@3.4.3:', type='build')
-    depends_on('pkgconfig',   type='build')
+    depends_on('pkgconfig', type='build')
 
-    depends_on('binutils')
-    depends_on('zlib')
-    # depends_on('unuran')
-    depends_on('freetype')
-    depends_on('pcre')
+    depends_on('lz4',    when='@6.13.02:')
+    depends_on('xxhash', when='@6.13.02:')
     depends_on('xz')
-    depends_on('libsm')
-    depends_on('libice')
-    depends_on('libx11')
-    depends_on('libxext')
-    depends_on('libxpm')
-    depends_on('libxft')
-    # depends_on('gif')
+    depends_on('pcre')
+    depends_on('freetype')
     depends_on('libpng')
-    depends_on('jpeg')
-    depends_on('gsl')
-    depends_on('python@2.7:')
-    # depends_on('opengl')
-    depends_on('graphviz', when='+graphviz')
-    # depends_on('kerberos')
-    depends_on('libxml2+python')
-    depends_on('openssl')
-    # depends_on('castor')
-    # depends_on('rfio')
-    # depends_on('mysql')
-    # depends_on('oracle')
-    # depends_on('odbc')
-    # depends_on('postgresql')
-    depends_on('sqlite')
-    # depends_on('pythia')
-    depends_on('fftw')
-    depends_on('cfitsio')
+    depends_on('ncurses')
+    depends_on('zlib')
+
+    # X-Graphics
+    depends_on('libx11',  when="+x")
+    depends_on('libxext', when="+x")
+    depends_on('libxft',  when="+x")
+    depends_on('libxpm',  when="+x")
+
+    # OpenGL
+    depends_on('ftgl',  when="+x+opengl")
+    depends_on('glew',  when="+x+opengl")
+    depends_on('gl',    when="+x+opengl")
+    depends_on('glu',   when="+x+opengl")
+    depends_on('gl2ps', when="+x+opengl")
+
+    # Qt4
+    depends_on('qt', when='+qt4')
```

Comment:
FYI the Spack Qt package now includes 5.x versions so you may want to constrain this